catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,81 +124,133 @@
 
   private boolean storeOriginalImage = true;
 
+  private CatalogFramework catalogFramework;
+
   static {
     IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
   }
 
   private double maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
 
   @Override
-  public CreateStorageRequest process(CreateStorageRequest createStorageRequest)
-      throws PluginExecutionException {
-    if (createStorageRequest == null) {
-      throw new PluginExecutionException(
-          "process(): argument 'createStorageRequest' may not be null.");
+  public CreateResponse process(CreateResponse createResponse) throws PluginExecutionException {
+    if (createResponse == null) {
+      throw new PluginExecutionException("process(): argument 'createResponse' may not be null.");
     }
-
-    process(createStorageRequest.getContentItems());
-    return createStorageRequest;
+    updateContent(
+        new HashSet<>(createResponse.getCreatedMetacards()),
+        createResponse.getRequest().getProperties());
+    return createResponse;
   }
 
   @Override
-  public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
-      throws PluginExecutionException {
-    if (updateStorageRequest == null) {
-      throw new PluginExecutionException(
-          "process(): argument 'updateStorageRequest' may not be null.");
+  public UpdateResponse process(UpdateResponse updateResponse) throws PluginExecutionException {
+    if (updateResponse == null) {
+      throw new PluginExecutionException("process(): argument 'updateResponse' may not be null.");
     }
+    updateContent(
+        updateResponse
+            .getUpdatedMetacards()
+            .stream()
+            .map(Update::getNewMetacard)
+            .collect(Collectors.toSet()),
+        updateResponse.getRequest().getProperties());
+    return updateResponse;
+  }
 
-    process(updateStorageRequest.getContentItems());
-    return updateStorageRequest;
+  @Override
+  public DeleteResponse process(DeleteResponse deleteResponse) throws PluginExecutionException {
+    return deleteResponse;
   }
 
-  private boolean isNitfMimeType(String rawMimeType) {
-    try {
-      return NITF_MIME_TYPE.match(rawMimeType);
-    } catch (MimeTypeParseException e) {
-      LOGGER.debug("unable to compare mime types: {} vs {}", NITF_MIME_TYPE, rawMimeType);
+  private void updateContent(Set<Metacard> metacards, Map<String, Serializable> properties) {
+    List<Metacard> metacardUpdates = new ArrayList<>();
+    List<ContentItem> contentUpdates = new ArrayList<>();
+    for (Metacard mcard : metacards) {
+      if (shouldGenerateContentItems(mcard, properties)) {
+        createContentItems(mcard, metacardUpdates, contentUpdates);
```

Comment:
It's a little misleading that this method is called `createContentItems` when it could only generate the thumbnail without creating any `ContentItems`. Maybe something like`generateImages` or `createUpdates` or would be better.